bundles/org.openhab.binding.opengarage/src/main/java/org/openhab/binding/opengarage/internal/OpenGarageHandler.java

```diff
@@ -53,33 +53,42 @@ public class OpenGarageHandler extends BaseThingHandler {
     private final Logger logger = LoggerFactory.getLogger(OpenGarageHandler.class);
 
     private long refreshInterval;
+    private long transitionTimeSecs;
 
     private @NonNullByDefault({}) OpenGarageWebTargets webTargets;
-    private @Nullable ScheduledFuture<?> pollFuture;
+    private @Nullable VariableDelayPoller poller;
+    private Instant lastTransition;
+    private String lastTransitionText;
+
+    public @NonNullByDefault({}) String doorOpeningState;
+    public @NonNullByDefault({}) String doorOpenState;
+    public @NonNullByDefault({}) String doorClosedState;
+    public @NonNullByDefault({}) String doorClosingState;
 
     public OpenGarageHandler(Thing thing) {
         super(thing);
+        this.lastTransition = Instant.MIN;
+        this.lastTransitionText = "";
     }
 
     @Override
     public void handleCommand(ChannelUID channelUID, Command command) {
         try {
             logger.debug("Received command {} for thing '{}' on channel {}", command, thing.getUID().getAsString(),
                     channelUID.getId());
-            boolean invert = isChannelInverted(channelUID.getId());
+            var maybeInvert = getInverter(channelUID.getId());
```

Comment:
I would prefer to avoid "var" in that place because with var you need to look at `getInverter` to guess the type.